src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php

```diff
@@ -540,6 +541,194 @@ public function findOneHost(int $hostId): ?Host
+            );
+            $services[] = $service;
+        }
+        return $services;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function findHostsByIdsForAdminUser(array $hostIds): array
+    {
+        $hosts = [];
+
+        if (empty($hostIds)) {
+            return $hosts;
+        }
+
+        $collector = new StatementCollector();
+
+        $request =
+            'SELECT h.*,
+            i.name AS poller_name,
+              IF (h.display_name LIKE \'_Module_Meta%\', \'Meta\', h.display_name) AS display_name,
+              IF (h.display_name LIKE \'_Module_Meta%\', \'0\', h.state) AS state
+            FROM `:dbstg`.`instances` i
+            INNER JOIN `:dbstg`.`hosts` h
+              ON h.instance_id = i.instance_id
+              AND h.enabled = \'1\'
+              AND h.name NOT LIKE \'_Module_BAM%\'';
+
+        if (is_array($hostIds)) {
+            $idsListKey = [];
+            foreach ($hostIds as $index => $id) {
+                $key = ":id{$index}";
+                $idsListKey[] = $key;
+                $collector->addValue($key, $id, \PDO::PARAM_INT);
+                unset($index, $id);
+            }
+            $request .= ' WHERE h.host_id IN (' . implode(',', $idsListKey) . ')';
+        }
+
+        $request = $this->translateDbName($request);
+
+        $statement = $this->db->prepare($request);
+        $collector->bind($statement);
+        $statement->execute();
+
+        while (false !== ($row = $statement->fetch(\PDO::FETCH_ASSOC))) {
+            $host = EntityCreator::createEntityByArray(
+                Host::class,
+                $row
+            );
+            $hosts[] = $host;
+        }
+
+        return $hosts;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function findHostsByIdsForNonAdminUser(array $hostIds): ?array
+    {
+        $hosts = [];
+
+        if (empty($hostIds)) {
+            return $hosts;
+        }
+
+        $collector = new StatementCollector();
+
+        $accessGroupFilter =
+            ' INNER JOIN `:dbstg`.`centreon_acl` acl
+                ON acl.host_id = h.host_id
+                AND acl.service_id IS NULL
+            INNER JOIN `:db`.`acl_groups` acg
+                ON acg.acl_group_id = acl.group_id
+                AND acg.acl_group_activate = \'1\'
+                AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups) . ') ';
+
+        $request =
+            'SELECT h.*,
+            i.name AS poller_name,
+              IF (h.display_name LIKE \'_Module_Meta%\', \'Meta\', h.display_name) AS display_name,
+              IF (h.display_name LIKE \'_Module_Meta%\', \'0\', h.state) AS state
+            FROM `:dbstg`.`instances` i
+            INNER JOIN `:dbstg`.`hosts` h
+              ON h.instance_id = i.instance_id
+              AND h.enabled = \'1\'
+              AND h.name NOT LIKE \'_Module_BAM%\''
+            . $accessGroupFilter;
+
+        if (is_array($hostIds)) {
```

Comment:
$hostIds is already checked as an array in the signature of this method
You can remove this condition.